App.js

```diff
@@ -18,11 +22,39 @@ const App = () => {
       'titillium-web-light': require('./assets/fonts/TitilliumWeb-Light.ttf'),
       'titillium-web-light-italic': require('./assets/fonts/TitilliumWeb-LightItalic.ttf')
     })
-      .then(() => setFontLoaded(true))
-      .catch((err) => console.warn('An error occurred with loading the fonts', err));
+      .catch((error) => console.warn('An error occurred with loading the fonts', error))
+      .finally(() => setFontLoaded(true));
   }, []);
 
   return fontLoaded ? <MainApp /> : null;
 };
 
+const App = () => {
+  const [matomoInstance, setMatomoInstance] = useState();
+
+  useEffect(() => {
+    SplashScreen.preventAutoHideAsync();
+
+    matomoSettings()
+      .then((settings) =>
+        setMatomoInstance(
+          new MatomoTracker({
+            urlBase: secrets[namespace].matomoUrl,
+            siteId: secrets[namespace].matomoSiteId,
+            userId: settings.userId,
+            disabled: !settings.consent || __DEV__,
+            log: __DEV__
+          })
+        )
+      )
+      .catch((error) => console.warn('An error occurred with setup Matomo tracking', error));
+  }, []);
+
+  return matomoInstance ? (
+    <MatomoProvider instance={matomoInstance}>
+      <AppWithFonts />
+    </MatomoProvider>
+  ) : null;
```

Comment:
Do we really want to have the complete app not being rendered when there is an error in the tracking setup?

This seems especially problematic since the way the effect is set up would mean that the user needs to force close the app in order to trigger a new attempt at setting up Matomo.

Reply:
done in https://github.com/ikuseiGmbH/smart-village-app-app/pull/109/commits/d92597a0bd1d840691b22f3258958008efba1689